target/linux/ar71xx/files/arch/mips/ath79/mach-rme-eg200.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * eTactica EG-200 board, based on 8devices Carambola2 module
+ *
+ * Copyright (C) 2015 Karl Palsson <karlp@etactica.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation.
+ */
+
+#include <asm/mach-ath79/ath79.h>
+#include <asm/mach-ath79/ar71xx_regs.h>
+#include "common.h"
+#include "dev-eth.h"
+#include "dev-gpio-buttons.h"
+#include "dev-leds-gpio.h"
+#include "dev-m25p80.h"
+#include "dev-usb.h"
+#include "dev-wmac.h"
+#include "machtypes.h"
+
+#define RME_EG200_GPIO_LED_WLAN			0
+#define RME_EG200_GPIO_LED_ETH0			13
+#define RME_EG200_GPIO_LED_ETACTICA		15
+#define RME_EG200_GPIO_LED_MODBUS		16
+
+#define RME_EG200_GPIO_BTN_RESTORE		11
+
+#define RME_EG200_KEYS_POLL_INTERVAL		20	/* msecs */
+#define RME_EG200_KEYS_DEBOUNCE_INTERVAL	(3 * RME_EG200_KEYS_POLL_INTERVAL)
+
+#define CARAMBOLA2_MAC0_OFFSET			0x0000
+#define CARAMBOLA2_CALDATA_OFFSET		0x1000
+#define CARAMBOLA2_WMAC_MAC_OFFSET		0x1002
+
+static struct gpio_led rme_eg200_leds_gpio[] __initdata = {
+	{
+		.name		= "eg200:red:wlan",
+		.gpio		= RME_EG200_GPIO_LED_WLAN,
+		.active_low	= 1,
+	}, {
+		.name		= "eg200:red:eth0",
+		.gpio		= RME_EG200_GPIO_LED_ETH0,
+		.active_low	= 1,
+	}, {
+		.name		= "eg200:red:etactica",
+		.gpio		= RME_EG200_GPIO_LED_ETACTICA,
+		.active_low	= 0,
+	}, {
+		.name		= "eg200:red:modbus",
+		.gpio		= RME_EG200_GPIO_LED_MODBUS,
+		.active_low	= 0,
+	}
+};
+
+static struct gpio_keys_button rme_eg200_keys[] __initdata = {
+	{
+		.desc		= "restore button",
+		.type		= EV_KEY,
+		.code		= KEY_WPS_BUTTON,
+		.debounce_interval = RME_EG200_KEYS_DEBOUNCE_INTERVAL,
+		.gpio		= RME_EG200_GPIO_BTN_RESTORE,
+		.active_low	= 1,
+	},
+};
+
+static void __init carambola2_common_setup(void)
+{
+	u8 *art = (u8 *) KSEG1ADDR(0x1fff0000);
+
+	ath79_register_m25p80(NULL);
+	ath79_register_wmac(art + CARAMBOLA2_CALDATA_OFFSET,
+			    art + CARAMBOLA2_WMAC_MAC_OFFSET);
+
+	ath79_setup_ar933x_phy4_switch(true, true);
+
+	ath79_init_mac(ath79_eth0_data.mac_addr, art + CARAMBOLA2_MAC0_OFFSET, 0);
+
+	ath79_register_mdio(0, 0x0);
+
+	/* WAN port */
+	ath79_register_eth(0);
+}
+
+static void __init rme_eg200_setup(void)
+{
+	carambola2_common_setup();
```

Comment:
why is this called carambola2_setup ?

Reply:
It looks like they reused code from Carambola 2 mach file since it is based on it.
https://github.com/openwrt/openwrt/blob/master/target/linux/ar71xx/files/arch/mips/ath79/mach-carambola2.c#L64

Reply:
Then it doesn't make sense to introduce another mach-*.c file for this device and include required code in mach-carambola2.c instead.

Reply:
Because it's a carambola2 module inside.  Until this all moves to DTS, I don't know a better way of saying that.  It should still say at the top of the file that it's built on a carambol2 module?

Reply:
> Because it's a carambola2 module inside. Until this all moves to DTS, I don't know a better way of saying that. It should still say at the top of the file that it's built on a carambol2 module?

I would do it this way:
- drop `mach-rme-eg200.c`
- drop `config ATH79_MACH_RME_EG200` from `Kconfig.openwrt` and kernel configs
- include `rme_eg200_leds_gpio[]`, `rme_eg200_leds_gpio[]`, `rme_eg200_setup(void)`, `MIPS_MACHINE(...)` and all required defines in `mach-carambola2.c`

Have a look how something similar is done in https://github.com/openwrt/openwrt/blob/master/target/linux/ar71xx/files/arch/mips/ath79/mach-archer-c7.c